derive_stringset/src/lib.rs

```diff
@@ -17,18 +17,17 @@ pub fn derive(input: TokenStream) -> TokenStream {
         unimplemented!();
     };
 
-    //let recurse_match_arms = fields.iter().filter(|f| {
-    //    use syn::Type;
-    //    match f {
-    //        Array => true,
-
-    //        _ => false
-    //    }
-
-    //})
-    
+    let recurse_match_arms = fields.iter().map(|f| {
+        let name = &f.ident;
+        quote! {
+            stringify!(#name) => {
+                self.#name.string_set(topic_parts, value)?;
```

Comment:
The base case shouldn't need to call `string_set`, since it just devolves down to a single call to `serde_json_core::from_str()`. Why not just call that directly here? That avoids us having to maintain the macro shenanigans for base types/arrays, since serde handles that for us automatically.